tools/java/org/unicode/cldr/api/FilteredData.java

```diff
@@ -0,0 +1,73 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package org.unicode.cldr.api;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Filters underlying {@link CldrData} source data to transform values or remove values on the fly.
+ * The order of transformed values produced by the class is the same order as the underlying data.
+ *
+ * <p>This class does not currently support injecting additional values due to the difficulty in
+ * ensuring correct ordering semantics.
+ *
+ * <p>A subclass should implement the {@link FilteredData#filter(CldrValue)} method to achieve the
+ * desired results. Note however that this API is currently unstable and is expected to change once
+ * DTD path ordering is enforced everywhere.
+ */
+// TODO: Once DTD ordering is the only allowed order, this can be extended to allow adding paths.
+public abstract class FilteredData implements CldrData {
+    private final CldrData src;
+
+    /** Constructs the parent class with a {@link CldrData} source. */
+    protected FilteredData(CldrData src) {
+        this.src = checkNotNull(src);
+    }
+
+    /** Returns the underlying source data to sub-class implementations. */
+    protected CldrData getSourceData() {
+        return src;
+    }
+
+    /**
+     * Returns a filtered CLDR value, replacing or removing the original value during visitation.
+     * The filtered value can only differ in it's base value or value attributes, and must have
```

Comment:
it's --> its

Reply:
not fixed yet

Reply:
Dammit, github does a spectacularly good job of hiding these comments when I look at the review. It's borderline impossible for me to find these things after a push of a later version of the code. Thanks for being patient.